src/uucore/src/lib/features/entries.rs

```diff
@@ -72,6 +72,31 @@ pub fn get_groups() -> IOResult<Vec<gid_t>> {
     }
 }
 
+pub fn get_groups_gnu(arg_id: Option<u32>) -> IOResult<Vec<gid_t>> {
+    // The list of group IDs return from GNU's `groups` and GNU's `id --groups`
+    // starts with the effective group ID (egid).
+    // This is a wrapper for `get_groups()` to mimic this behavior.
+    //
+    // If `arg_id` is None (default), `get_groups_gnu` moves the effective
+    // group id (egid) to the first entry in the returned Vector.
+    // If `arg_id` is Some(x), `get_groups_gnu` moves the id with value `x`
+    // to the first entry in the returned Vector. This might be necessary
+    // for `id --groups --real` if `gid` and `egid` are not equal.
```

Comment:
Great explanation! It would be great to turn it into a rustdoc comment for this function!